pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -359,7 +359,7 @@ func getTagListFromSpec(spec *Spec) map[string]*string {
 
 	tagList := map[string]*string{}
 	for key, element := range spec.Tags {
-		tagList[key] = to.StringPtr(element)
+		tagList[key] = element
```

Comment:
getTagListFromSpec() method will not be required if the Tag type(`type AzureMachineProviderSpec struct` of openshift/api) is updated to `map[string]*string` as defined in azure-sdk.